command-based/robot.py

```diff
@@ -23,14 +24,16 @@ def robotInit(self):
         you will need to access later.
         '''
 
-        subsystems.init()
+        Command.getRobot = lambda x=0: self
```

Comment:
I like this solution -- what if you just moved it to your commandbasedrobot.py in robotpy-wpilib-utilities? Then it wouldn't need to be copy/pasted to every new command based robot.

Also, it might be nicer to attach a property instead -- then you could do `self.robot` in commands.

Reply:
Attaching a property to a class creates a static variable, which #5 says is as bad as a global and should be removed from all of WPILib.

Reply:
Attaching a property isn't any different than attaching a function -- underneath the hood a property is just a bunch of function calls anyways. :)

Reply:
Also, if you attach it in the commandbasedrobot.py, then you can attach a reset function which will delete it when tests reset, which would be a bonus. 👍

Reply:
A `Command` having a `robot` property doesn't really seem like the right abstraction. A command (or all commands) doesn't have a robot. Commands control the robot's subsystems, which is why we use modules to keep them separate.

What would a reset function include that wouldn't be accomplished by re-running `robotInit()`?

Reply:
Well, the reset function I think is called on test destruction, and it's guaranteed to be called... it wouldn't be a huge improvement, but it would lessen the chance that a stale robot object was being used? And feels cleaner.

If you prefer getRobot over robot, that's fine. I think I have a strong preference for putting that in commandbasedrobot.py -- but if you think it belongs here, I'm fine with merging this as is. Thanks for your continued contributions.

Reply:
Is there an example `reset()` function we could look at? We didn't find any attached to the other robot base classes.

Reply:
https://github.com/robotpy/robotpy-wpilib/blob/84852422f2a53f098ba3ad05db74075f4ef93719/wpilib/wpilib/smartdashboard.py#L40

Reply:
Okay, that's interesting. At the moment it doesn't seem useful for `CommandBasedRobot` because the only variable it could reset is the Scheduler instance, which already has its own `_reset()` method.

We're still considering our options for more pythonic subsystems. If we find an answer that we think is the [Right Thing](http://www.catb.org/jargon/html/R/Right-Thing.html) and universally applicable, we'll update the `CommandBasedRobot` class and this example.